File: app/src/main/java/br/odb/angstronme/Player.java
package br.odb.angstronme;

/**
 * Created by monty on 6/6/15.
 */
public class Player {

	public final Team team;
	public final Vec3 position = new Vec3(0, 0, 0);
	public Directions direction;

	public Player(Team team) {
		this.team = team;
	}

	public void updatePosition() {
		switch (direction) {
			case N:
				position.y--;
				break;
			case E:
				position.x--;
				break;
			case S:
				position.y++;
				break;
			case W:
				position.x++;
				break;
		}
	}
	public enum Team {NOTHING, PLAYER, Team, CPU}

	public enum Directions {N, E, S, W}
}

File: app/src/main/java/br/odb/angstronme/Player.kt
package br.odb.angstronme

/**
 * Created by monty on 6/6/15.
 */
class Player {
	@JvmField
	val position = Vec3(0.0f, 0.0f, 0.0f)
	@JvmField
	var direction: Directions? = null
	fun updatePosition() {
		when (direction) {
			Directions.N -> position.y--
			Directions.E -> position.x--
			Directions.S -> position.y++
			Directions.W -> position.x++
			else -> {}
		}
	}

	enum class Team {
		NOTHING, PLAYER, CPU
	}

	enum class Directions {
		N, E, S, W
	}
}